### PR TITLE
Add udr challenge type

### PR DESCRIPTION
**Pull request by jan-win1993**
Add new ACME challenge tyoe:

- udr

refs https://github.com/opnsense/plugins/issues/3035

### CODE REVIEW

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml [hunk 1]
```diff
@@ -1098,6 +1098,21 @@
         <type>textbox</type>
         <help>Requires the whole key file in a format that is compatible with TransIP.</help>
     </field>
+    <field>
+        <label>udr</label>
```
Comment (fraenki) [inline security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml:1102]: This is a bit cryptic for a label. Would you mind to relabel it to "united-domains Reselling"? :)

**Review by fraenki — CHANGES_REQUESTED** [review]

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml (hunk 1 above)
Comment (jan-win1993) [reply security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml]: Done :)

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml [hunk 2]
```diff
@@ -1098,6 +1098,21 @@
...
+    </field>
+    <field>
+        <id>validation.dns_udr_user</id>
+        <label>User</label>
+        <type>text</type>
+    </field>
+    <field>
+        <id>validation.dns_udr_password</id>
+        <label>Password</label>
+        <type>text</type>
```
[the remark this answers is not in the document]
Comment (jan-win1993) [reply security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/dialogValidation.xml]: Of course :)

**Comment by fraenki** [comment]
Merged, thanks!

**Comment by ChristopherZender** [comment]
thanks!

**Pull request merged**